modules/nf-core/samtools/merge/main.nf

```diff
@@ -7,17 +7,18 @@ process SAMTOOLS_MERGE {
         'https://depot.galaxyproject.org/singularity/samtools:1.20--h50ea8bc_0' :
         'biocontainers/samtools:1.20--h50ea8bc_0' }"
 
+    // arity cannot be used on optional input and output paths: https://github.com/nextflow-io/nextflow/issues/5111
     input:
-    tuple val(meta), path(input_files, stageAs: "?/*")
+    tuple val(meta), path(input_files, stageAs: "?/*", arity: '1..*') // One or more files are expected
     tuple val(meta2), path(fasta)
     tuple val(meta3), path(fai)
 
     output:
-    tuple val(meta), path("${prefix}.bam") , optional:true, emit: bam
-    tuple val(meta), path("${prefix}.cram"), optional:true, emit: cram
-    tuple val(meta), path("*.csi")         , optional:true, emit: csi
-    tuple val(meta), path("*.crai")        , optional:true, emit: crai
-    path  "versions.yml"                                  , emit: versions
+    tuple val(meta), path("${prefix}.bam")    , optional:true, emit: bam
+    tuple val(meta), path("${prefix}.cram")   , optional:true, emit: cram
+    tuple val(meta), path("*.csi")            , optional:true, emit: csi
+    tuple val(meta), path("*.crai")           , optional:true, emit: crai
+    path  "versions.yml"                      , emit: versions
```

Comment:
I think the previous way is actually more harshil conform

Reply:
How so? :grin: